launchcontainer/launchcontainer.py

```diff
@@ -68,13 +66,53 @@ def student_email(self):
         else:
             return None
 
-    def _get_API_url(self):
-        uri = settings.ENV_TOKENS.get('LAUNCHCONTAINER_API_CONF', DEFAULT_API_CONF)
+    def _validate_conf(self, uri):
+        """Receive a LAUNCHCONTAINER_API_CONF dict and ensure that it is a valid URI.
+
+        This method expects the configuration param to follow this format:
+        """
+
         url_validator = validators.URLValidator()
-        url_validator(uri)
+        try:
+            url_validator(uri)
+        except validators.ValidationError:
+            msg = ("The LAUNCHCONTAINER_API_CONF is not configured properly. "
+                   "Please provide a valid URL.")
+            logger.warning(msg)
+            return None
 
         return uri
 
+    def _get_API_url(self):
+        """Check to see if the URL has been configured in the microsite. If not,
+        fall back to the settings. Finally, default to whatever is defined here."""
+
+        DEFAULT_API_CONF = 'https://wharf.appsembler.com/isc/newdeploy/'
+
+        api_conf = {}
+
+        # Use the microsite value if it is available.
+        if hasattr(self, "runtime"):
+            # TODO: We should really be validating this as the JSON is submitted
+            # in the admin. Let's look into that soon.
+            organization = self.runtime.course_id.org
+            microsite_query = Microsite.objects.filter(key=organization)
```

Comment:
What about doing something like:

```python
try:
    microsite = Microsite.objects.get(key=organization)
except Microsite.DoesNotExist:
    pass
else:
    api_conf = microsite.values.get('LAUNCHCONTAINER_API_CONF')

# (...)

api_conf = self._validate_conf(api_conf)
return api_conf
```

(Note that this will also fail when there are multiple Microsites with the same `key`--not sure if that's what we want or not?)